airflow/models.py

```diff
@@ -346,10 +346,10 @@ def kill_zombies(self, session=None):
 
                     # now set non db backed vars on ti
                     ti.task = task
-                    ti.test_mode = False
+                    ti.test_mode = configuration.getboolean('core', 'unit_test_mode')
```

Comment:
@bolkedebruin can we take a decision on this? As per my understanding it should be always `False` here , otherwise that `TI` won't end up in `kill_zombie` method. 

@mistercrunch  @r39132 @criccomini can you guys throw some more light on it.